macros/upload_results.sql

```diff
@@ -78,14 +78,23 @@
 
         {% do log("Uploading tests", true) %}
         {% set tests = dbt_artifacts.get_relation('tests') %}
-        {% set content_tests = dbt_artifacts.upload_tests(graph) %}
-        {{ dbt_artifacts.insert_into_metadata_table(
-            database_name=tests.database,
-            schema_name=tests.schema,
-            table_name=tests.identifier,
-            content=content_tests
-            )
-        }}
+        {% set tests_set = [] %}
+        {% for node in graph.nodes.values() | selectattr("resource_type", "equalto", "test") %}
+            {% do tests_set.append(node) %}
+        {% endfor %}
+        {# upload tests in chunks of 5000 tests, or less #}
+        {% set upload_limit = 5000 %}
+        {% set n = (tests_set|length/upload_limit)|round(0, 'ceil')|int %}
+        {% for i in range(n) -%}
+            {% set content_tests = upload_tests(tests_set, i+1, upload_limit) %}
```

Comment:
I believe this is why the integration tests are currently failing:

```suggestion
            {% set content_tests = dbt_artifacts.upload_tests(tests_set, i+1, upload_limit) %}
```